### PR TITLE
Enable testing using remote processes in uapaot

### PR DESCRIPTION
**Pull request by alexperovich**
This change enables using the RemoteExecutorConsoleApp on AOT by
compiling each executable in the input directory and merging the
outputs.

### CODE REVIEW

File: Tools-Override/tests.targets [hunk 1]
```diff
@@ -232,9 +232,15 @@
       <IlcInputFolderContents Include="$(ILCFXInputFolder)/*" />
       <TestCommandLines Include="mklink /H %(IlcInputFolderContents.Filename)%(IlcInputFolderContents.Extension) $(_Runtime_Path)%(IlcInputFolderContents.Filename)%(IlcInputFolderContents.Extension)" />
       <TestCommandLines Include="copy /y $(_TestILCFolder)\default.rd.xml  %EXECUTION_DIR%" />
-      <TestCommandLines Include="call $(_TestILCFolder)\ilc.exe -ExeName xunit.console.netcore.exe -in %EXECUTION_DIR% -out %EXECUTION_DIR%\native -usedefaultpinvoke -buildtype $(ILCBuildType) -v diag" />
-      <TestCommandLines Include="set ILCERRORLEVEL=%ERRORLEVEL%" />
-      <TestCommandLines Include="if NOT [%ILCERRORLEVEL%] == [0] exit /b %ILCERRORLEVEL%" />
+      <TestCommandLines Include="robocopy /MOV /NP %EXECUTION_DIR% %EXECUTION_DIR%\ilc.input /XF %~nx0" />
```
Comment (AtsushiKan) [inline Tools-Override/tests.targets]: ilc.input sounds too much like a filename with an extension. Can it be renamed to ilc_input or ilcinput?

File: Tools-Override/tests.targets [hunk 2]
```diff
@@ -232,9 +232,15 @@
...
+      <TestCommandLines Include="setlocal enabledelayedexpansion" />
```
Comment (AtsushiKan) [inline Tools-Override/tests.targets]: The "/MOV" flag to robocopy makes me suspicious. RunTests.cmd is supposed to be repeatable...

File: Tools-Override/tests.targets [hunk 3]
```diff
@@ -232,9 +232,15 @@
...
+      <TestCommandLines Include="for /r &quot;%EXECUTION_DIR%\ilc.input&quot; %%f in (*.exe) do (" />
```
Comment (AtsushiKan) [inline Tools-Override/tests.targets]: Do we want to ILC every .exe file indiscriminately? What if a test drops a native .exe as a test resource in its execution directory?

**Comment by AtsushiKan** [comment]
How did you solve the problem of ILC not knowing which assembly(s) RemoteExecutorConsoleApp is going to load?

**Comment by alexperovich** [comment]
I assume that RemoteExecutorConsoleApp is going to only load assemblies in the input folder. The default.rd.xml we have tells ILC to include all assemblies in the application.

File: Tools-Override/tests.targets [hunk 4]
```diff
@@ -229,13 +229,20 @@
...
       <TestCommandLines Include="mklink /H %(IlcInputFolderContents.Filename)%(IlcInputFolderContents.Extension) $(_Runtime_Path)%(IlcInputFolderContents.Filename)%(IlcInputFolderContents.Extension)" />
       <TestCommandLines Include="copy /y $(_TestILCFolder)\default.rd.xml  %EXECUTION_DIR%" />
-      <TestCommandLines Include="call $(_TestILCFolder)\ilc.exe -ExeName xunit.console.netcore.exe -in %EXECUTION_DIR% -out %EXECUTION_DIR%\native -usedefaultpinvoke -buildtype $(ILCBuildType) -v diag" />
-      <TestCommandLines Include="set ILCERRORLEVEL=%ERRORLEVEL%" />
-      <TestCommandLines Include="if NOT [%ILCERRORLEVEL%] == [0] exit /b %ILCERRORLEVEL%" />
-      <TestCommandLines Include="copy /y $(_TestILCFolder)\CRT\vcruntime140_app.dll %EXECUTION_DIR%\native" />
+      <TestCommandLines Include="rmdir /S /Q %EXECUTION_DIR%int" />
+      <TestCommandLines Include="rmdir /S /Q %EXECUTION_DIR%native" />
+      <TestCommandLines Include="@(TargetExecutableNames -> '
+call %RUNTIME_PATH%\TestILC\ilc.exe -ExeName %(Identity) -in %EXECUTION_DIR% -out %EXECUTION_DIR%int\%(Identity)\ -usedefaultpinvoke -buildtype $(ILCBuildType) -v diag
```
Comment (weshaggard) [inline Tools-Override/tests.targets]: LOL

**Pull request merged**